README
Autor: Robson Vaamonde
Site: www.procedimentosemti.com.br
Facebook: facebook.com/ProcedimentosEmTI
Facebook: facebook.com/BoraParaPratica
YouTube: youtube.com/BoraParaPratica
Data de criação: 31/05/2016
Data de atualização: 26/09/2016
Versão: 0.6
Testado e homologado para a versão do Ubuntu Server 16.04 LTS x64
Kernel >= 4.4.x

Pasta doc:	Padronização de Partições e Diretórios.pdf
			Padronização De Nomes de Host e Server.pdf

Procedimento de utilização dos scripts de instalação e configuração do SAMBA4 no GNU/Linux Ubuntu Server 16.04 x64 LTS.
Sequência: script-00.sh até script-20.sh, alguns scripts vão fazer o reboot automático do servidor.

# Padronização da instalação do servidor: ptispo01dc01
# Criação da estrutura de discos
/dev/sda	- OS	-	 8GB	RAID1	LVM
/dev/sdb	- OS	-	 8GB	RAID1	LVM
/dev/sdc	- Files	-	10GB	RAID1	LVM
/dev/sdd	- Files	-	10GB	RAID1	LVM
/dev/sde	- Bkp	-	15GB	SRAID	LVM

# Criação da estrutura do LVM
Grupos LVM:
root		/			-	4.0GB		-	Ext4	-	Defaults,Barrier=1
			/tmp		-	512MB		-	Ext4	-	Noexec, Nosuid, Nodev
			/swap		-	512MB		-	Swap
			/var		-	3.0GB		-	Ext4	-	Defaults,Barrier=1
files		/arquivos	-	 10GB		-	Ext4	-	Defaults,Barrier=1,Grpquota,Usrquota
bkp			/backup		-	 15GB		-	Ext4	-	Defaults,Barrier=1

# Padronização da raiz de diretórios:
/arquivos
		/pti.intra
				/publica		Letra: P: (quota de disco);
				/sistema		Letra: S:
				/gestao			Letra: G:
				/usuarios		Letra: U: (quota de disco);
						/perfil
								/documentos
/backup
		/pti.intra
				/samba-ad-dc
				/backup
				/rsync
					/sistema
					/gestao
					/usuarios
					/backup

Script-00.sh: Script responsavél por fazer as atualização do sistema
# Instalação dos pacotes principais para a primeira etapa, indicado para a distribuição GNU/Linux Ubuntu Server 16.04 LTS x64
#
# Atualização das listas do Apt-Get
# Atualização dos Aplicativos Instalados
# Atualização da Distribuição Ubuntu Server (Kernel)
# Auto-Limpeza do Apt-Get
# Limpando o repositório Local do Apt-Get

Script-01.sh: Script responsável por instalar as dependências de rede, configurações do Kerberos e FSTAB
# Instalação dos pacotes principais para a segunda etapa, indicado para a distribuição GNU/Linux Ubuntu Server 16.04 LTS x64
#
# NTP (Network Time Protocol) Servidor de Data/Hora
# KRB5 (Kerberos) Protocolo de Autenticação Segura
# NFS (Network File System) Protocolo de Transferência de Arquivos
# ACL (Access Control List) Permissões de Arquivos e Diretórios
# ATTR (Extended Attributes) Atributos Extendidos
#
# Após o reboot fazer as mudanças do arquivo /etc/fstab para suportar os recursos de ACL e XATTR EXT4
#	vim /etc/fstab
#	defaults,barrier=1
#
# Se tiver utilizando o sistema de arquivos BTRFS, deixar o padrão
#	vim /etc/fstab
#	defaults,subvol=@
#
# Após o reboot configurar o arquivo /etc/ntp.conf para atualizar data e hora dos servidores do NTP.br
#	 mv /etc/ntp.conf /etc/ntp.conf.old
#	 cp ntp.drift /etc/
#	 cp ntp.conf /etc/
#
# Na instalação fazer a criação do REALM do Kerberos
#	REALM: 	PTI.INTRA
#	SERVER:	ptispo01dc01.pti.intra	
#	ADMIN:	ptispo01dc01.pti.intra
#	debconf-show krb5-config

Script-02.sh: Script responsavél por instalar todos os pacotes para o servidor de arquivos SAMBA4 e recursos de Rede
# Instalação dos pacotes principais para a terceira etapa, indicado para a distribuição GNU/Linux Ubuntu Server 16.04 LTS x64
#
# SAMBA4 (Server Message Block) Serviço de Armazenamento e Gerenciamento de Arquivos e Usuários
# DNS (Domain Name System) Serviço de Domínio de Nomes
# CUPS (Common Unix Printing System) Serviços de Impressão
# DHCP (Dynamic Host Configuration Protocol) Configuração Dinâmica de Computadores
# WINBIND Integração SAMBA + Linux
# QUOTA Criação de Quotas de Discos
#
# Após o reboot fazer as mudanças do arquivo /etc/nsswitch.conf para suportar a autenticação via Winbind
#	vim /etc/nsswitch.conf
#	passwd:		files	compat	winbind
#	group:		files	compat	winbind
#	shadown:	files	compat	winbind
#	hosts:		files 	dns   mdns4_minimal [NOTFOUND=return]
#
# Configurações de suporte a Quota, Acl e Xattr no rquivo /etc/fstab
# Utilizar essas configuração apenas para o sistema de arquivos EXT4
# Se tiver utilizando o BTRFS deixar o padrão
#
#	vim /etc/fstab
#	defaults,barrier=1,grpquota,usrquota
#
# Comando para confirmar as modificações feitas no FSTAB
#	tune2fs -l /dev/sda6 | grep "Default mount options:"
#
# Editando o arquivo /etc/hostname para acrescentar o FQDN
#	vim /etc/hostname
#	ptispo01dc01.pti.intra
#
# Editando o arquivo /etc/hosts
#	vim /etc/hosts
#	192.168.1.10	ptispo01dc01.pti.intra	ptispo01dc01
#
# Editando o arquivo /etc/defaults/grub
#	GRUB_CMDLINE_LINUX_DEFAULT="net.ifnames=0" 
#	update-grub
#
# Atualizando o ClamAV
#	freshclam
#	service clamav-daemon start
#	service clamav-freshclam start

Script-03.sh: Script responsavél pela instalação do Webmin
# Instalação dos pacotes principais para a quarta etapa, indicado para a distribuição GNU/Linux Ubuntu Server 16.04 LTS x64
#
# Instalando as dependências para Webmin - WebAdmin
# Baixando o Webmin do site Oficial (versão: 1.801)
# Instalando o Webmin via dpkg
# Porta padrão de acesso ao Webmin: https://SEU_ENDEREÇO_IP:10000

Script-04.sh: Script responsável pela instalação do LAMP-SERVER
# Instalação dos pacotes principais para a quinta etapa, indicado para a distribuição GNU/Linux Ubuntu Server 16.04 LTS x64
#
# APACHE (Apache HTTP Server) -Servidor de Hospedagem de Páginas web
# MYSQL (SGBD) - Sistemas de Gerenciamento de Banco de Dados
# PHP (Personal Home Page - PHP: Hypertext Preprocessor) - Linguagem de Programação Dinâmica para Web
# PERL - Linguagem de programação multiplataforma
# PYTHON - Linguagem de programação de alto nível
# PhpMyAdmin - Aplicativo desenvolvido em PHP para administração do MySQL pela Internet
# PhpLDAPAdmin - Aplicativo desenvlvido em PHP para administração do LDAP SAMBA4 pela Internet
#
# Configuração do MySQL
#	será solicitado a senha do ROOT do MySQL
#
# Configuração do ProFTPD
#	será solicitado o tipo de inicialização do sistema, escolha: standalone
#
# Configuração do PhpMyAdmin
#	será solicitado o tipo do servidor web, escolha: apache2
#	será solicitado para configurar a base de dados do PhpMyAdmin, escolha: yes
#	será solicitado a senha do PhpMyAdmin, digite uma senha

Script-05.sh: Script responsável pela configuração dos serviços de Rede
# Configuração dos serviços da sexta etapa, indicado para a distribuição GNU/Linux Ubuntu Server 16.04 LTS x64
#
# Configuração do arquivo interfaces
#	vim /etc/network/interfaces
#
# Configuração do Serviço de SSHD, arquivo sshd_config
#	vim /etc/ssh/sshd_config
#
# Configuração dos arquivos hosts.allow e hosts.deny
#	vim /etc/hosts.allow
#	vim /etc/hosts.deny
#
# Configuração do Serviço de DHCPD, arquivo dhcpd.conf
#	vim /etc/dhcp/dhcpd.conf

Script-06.sh: Script responsável pela pormoção do Servidor SAMBA4 como Controlador de Domínio
# Configuração dos serviços do SAMBA4 da sétima etapa, indicado para a distribuição GNU/Linux Ubuntu Server 16.04 LTS x64
#
# Opções do samba-tool domain provision
#

CHANGELOG
Autor: Robson Vaamonde
Site: www.procedimentosemti.com.br
Facebook: facebook.com/ProcedimentosEmTI
Facebook: facebook.com/BoraParaPratica
YouTube: youtube.com/BoraParaPratica
Data de criação: 31/05/2016
Data de atualização: 17/07/2019
Versão: 0.48
Testado e homologado para a versão do Ubuntu Server 16.04 LTS x64
Testado e homologado para a versão do Ubuntu Server 16.04.5 LTS x64
Testado e homologado para a versão do Ubuntu Server 16.04.6 LTS x64
Kernel >= 4.4.x

17/07/2019 - Robson Vaamonde. Melhoria no script-06.sh, quebra de linhas na promoção do domínio e correções de documentação.

15/07/2019 - Robson Vaamonde. Correção do arquivo fstab, erro na linha da partição /tmp, falta de uma vírgula.
15/07/2019 - Robson Vaamonde. Melhoria no script-01.sh, opção de quebra de linha na instalação de pacotes utilizandoo apt-get
install \ "contra barra".
15/07/2019 - Robson Vaamonde. Atualização do script-02.sh, atualização das linhas do wget do EICAR, mudança do protocolo HTTP
para HTTPS.
15/07/2019 - Robson Vaamonde. Atualização do script-03.sh, opção de quebra de linha na instalação de pacotes utilizandoo apt-get
install \ "contra barra", atualização da versão do Webmin para 1.920
15/07/2019 - Robson Vaamonde. Atualização do script-20.sh, atualização da dependência de monitoramente do Netdata: mysql-client 
python-pymssql python-mysqldb, criação dos arquivos de monitoramento do MySQL e do DHCP Server.
15/07/2019 - Robson Vaamonde. Criação dos arquivos de monitoramento do MySQL mysql_srv.conf e DHCP Server isc_dhcpd.conf utilizados
pelo Netdata.
15/07/2019 - Robson Vaamonde. Atualização do script-24.sh, atualização da versão do LogAnalyzer para 4.1.7

09/10/2018 - Robson Vaamonde. Alteração das configurações do AWStats, para desabilitar o envio de email para root. 
Conf: https://github.com/vaamonde/samba4-l2/blob/master/conf/awstats <-- desabilitar a variável: MAILTO="" acréscimo da linha
de desvio de mangens para /dev/null
Script: https://github.com/vaamonde/samba4-l2/blob/master/script-22.sh <-- acréscimo das linhas de atualização do arquivo 
awstats

09/10/2018 - Robson Vaamonde. Alteração das configurações dos servidores de Bind, Grub e Firewall.
Conf: https://github.com/vaamonde/samba4-l2/blob/master/conf/named.conf.options <-- adicionado opção de logs e estatísticas do bind
Conf: https://github.com/vaamonde/samba4-l2/blob/master/conf/grub <-- adicionado a opção: consoleblank=0
Conf: https://github.com/vaamonde/samba4-l2/blob/master/conf/firewall <-- adicionado a opção: echo 1 > /sys/block/sda/device/queue_depth
Script: https://github.com/vaamonde/samba4-l2/blob/master/script-06.sh <-- criação do diretório/arquivo de estatísticas do bind rndc

09/10/2018  - Alteração das configurações de Firewall para liberação das portas TCP/UDP 514 do Rsyslog. Robson Vaamonde
Conf: https://github.com/vaamonde/samba4-l2/blob/master/conf/portslibtcp
Conf: https://github.com/vaamonde/samba4-l2/blob/master/conf/portslibudp

07/10/2018  - Acréscimo do monitoramento dos serviços do Rsyslog e Winbind no script-11.sh. Robson Vaamonde
Comando: sudo service rsyslog status
Comando: sudo service winbind status
Conf: https://github.com/vaamonde/samba4-l2/blob/master/script-11.sh

06/10/2018  - Correções das falhas dos script-24.sh e script-25.sh, preparação da gravação das vídeos aulas. Robson Vaamonde
(AULAS GRAVAS E JÁ DISPONIBILIZADA NO AULAEAD.COM NO DIA 08/10/2018)

06/10/2018  - Identificado pelos alunos falha no script-15.sh, erro de execução na linha 84. Robson Vaamonde
Falha: linha 84 do script: ldbsearch -H $SAMBA4LDB | grep "dn: OU" | | cat -n | less (tinha dois Pipes)

script-06.sh
#!/bin/bash
# Autor: Robson Vaamonde
# Site: www.procedimentosemti.com.br
# Facebook: facebook.com/ProcedimentosEmTI
# Facebook: facebook.com/BoraParaPratica
# YouTube: youtube.com/BoraParaPratica
# Data de criação: 31/05/2016
# Data de atualização: 17/07/2019
# Versão: 0.12
# Testado e homologado para a versão do Ubuntu Server 16.04 LTS x64
# Kernel >= 4.4.x
#
# Configuração dos serviços do SAMBA 4 da sétima etapa, indicado para a distribuição GNU/Linux Ubuntu Server 16.04 LTS x64
#
# Opções do samba-tool domain provision
#
# --option=”interfaces=lo eth0″ --option=”bind interfaces only=yes” – Se seu controlador de domínio tem mais de uma 
# interface de rede essa opção é obrigatória. Isso força o SAMBA 4 a escutar e resolver corretamente o DNS para a interface
# da sua rede local;
#
# --option="allow dns updates = nonsecure and secure" 
# --option="dns forwarder = 192.168.1.10"
# --option="winbind use default domain = yes" 
# --option="winbind enum users  = yes" 
# --option="winbind enum groups = yes" 
# --option="winbind refresh tickets = yes" 
# --option="winbind refresh tickets = rfc2307"
# --option="server signing = auto" 
# --option="vfs objects = acl_xattr" 
# --option="map acl inherit = yes" 
# --option="store dos attibutes = yes" 
# --option="client use spnego = no"
#
# --use-rfc2307 – O uso dessa opção ativa o SAMBA 4 Active Directory para armazenar atributos posix. É também cria
# informações do NIS (Network Information Service) no AD, permitindo a administração de UIDs/GIDs e outras funções Unix;
#
# --realm=PTI.INTRA – É onde setamos o domínio completo, no caso prefixo mais sufixo, obrigatório ser em MAIUSCULA;
#
# --domain=PTI – Nessa opção, devemos setar apenas o prefixo do domínio, obrigatório ser em MAIUSCULA;
#
# --dns-backend=BIND9_DLZ – Por padrão SAMBA 4 utiliza o seu próprio DNS. Essa opção força o SSAMBA 4 utilizar o Bind como DNS 
# Backend.
#
# --adminpass=’pti@2016’ – Aqui configuramos a senha do usuário “Administrator”, que será o administrador do sistema.
#
# --host-ip=192.168.1.10 - Aqui configuramos o endereço IP do servidor do SAMBA 4
#
# Informações sobre tipos de DNS do SAMBA 4
# SAMBA_INTERNAL	= Configurações de DNS interna no SAMBA 4
# BIND9_FLATFILE	= Configurações de DNS utilizando arquivos de configurações
# BIND9_DLZ		= Configurações de DNS integrado em SAMBA 4 é o BIND
# NONE			= Sem configuração de DNS
#
# Versões do Schema do Active Directory suportador pelo SAMBA 4
#
# Informações sobre o FSMO Roles (Flexible Single Master Operation Roles)
# 1. Schema Master (Mestre de Esquema);
# 2. Domain Name Master (Mestre de Nomeação de Domínio);
# 3. PDC Emulator (Emulador de Controlador de Domínio Primário);
# 4. RID Master (Mestre ID Relativo - SID);
# 5. Infra-Structure Master (Mestre de Infra-Estrutura);
# 6. GC Global Catalog (Catalog Global).
# Mais informações: https://wiki.samba.org/index.php/AD_Schema_Version_Support
#
# AppArmor configurações:
# r = permissões de leitura
# w = permissões de gravação
# k = permissões de lock
# m = ativação da flag PROT_EXEC
# Ux = permissão de execução de scripts
# Mais informações: http://manpages.ubuntu.com/manpages/xenial/en/man5/apparmor.d.5.html
#
# Utilizar o comando: sudo -i para executar o script
#
# Caminho para o Log do Script-06.sh
LOG="/var/log/script-06.log"
#
# Variável da Data Inicial para calcular tempo de execução do Script
DATAINICIAL=`date +%s`
#
# Validando o ambiente, verificando se o usuário e "root"
USUARIO=`id -u`
UBUNTU=`lsb_release -rs`
KERNEL=`uname -r | cut -d'.' -f1,2`

if [ "$USUARIO" == "0" ]
then
	if [ "$UBUNTU" == "16.04" ]
		then
			if [ "$KERNEL" == "4.4" ]
				then		
					 clear
					 # Variáveis de configuração do SAMBA 4
					 REALM="PTI.INTRA"
					 DOMAIN="PTI"
					 ROLE="dc"
					 DNS="BIND9_DLZ"
					 PASSWORD="pti@2016"
					 LEVEL="2008_R2"
					 SITE="PTI.INTRA"
					 IP="192.168.1.10"
					 #
					 echo -e "Usuário é `whoami`, continuando a executar o Script-06.sh"
					 echo
					 echo -e "Promovendo o SAMBA 4 como Controlador de Domínio"
					 echo
					 echo -e "Configurando o SAMBA 4 integrado com o BIND9-DNS-SERVER e ISC-DHCP-SERVER"
					 echo
					 echo -e "Nível Funcional do SAMBA 4 como Windows Server 2008 R2"
					 echo
					 echo -e "1. Schema Master (Mestre de Esquema);"
					 echo -e "2. Domain Name Master (Mestre de Nomeação de Domínio);"
					 echo -e "3. PDC Emulator (Emulador de Controlador de Domínio Primário);"
					 echo -e "4. RID Master (Mestre ID Relativo - SID);"
					 echo -e "5. Infra-Structure Master (Mestre de Infra-Estrutura);"
					 echo -e "6. GC Global Catalog (Catálogo Global)."
					 echo
					 echo -e "Aguarde..."
					 echo
					 echo -e "Rodando o Script-06.sh em: `date`" > $LOG
					 
					 echo -e "Atualizando as Listas do Apt-Get, aguarde..."
					 # Exportando o recurso de Noninteractive do Debconf
					 export DEBIAN_FRONTEND=noninteractive
					 #Atualizando as listas do apt-get
					 apt-get update &>> $LOG
					 echo -e "Listas Atualizadas com Sucesso!!!, continuando o script..."
					 echo
					 echo ============================================================ >> $LOG

					 echo -e "Atualizando o Sistema, aguarde..."
					 #Fazendo a atualização de todos os pacotes instalados no servidor
					 apt-get -o Dpkg::Options::="--force-confold" upgrade -q -y --force-yes &>> $LOG
					 echo -e "Sistema Atualizado com Sucesso!!!, continuando o script..."
					 echo
					 echo ============================================================ >> $LOG
					 
					 echo -e "Serviços atualizados com sucesso!!!, pressione <Enter> para continuar com o script"
					 read
					 sleep 2
					 clear
					 echo ============================================================ >> $LOG

					 echo -e "Promovendo o Servidor SAMBA 4 como AD-DC, pressione <Enter> para iniciar o processo de promoção"
					 echo 
					 read
					 echo
					 
					 echo -e "Parando o serviço do SAMBA 4, aguarde..."
					 #Parando o serviço do SAMBA 4
					 sudo service samba stop &>> $LOG
					 echo -e "Serviço parado com sucesso!!!, continuando o script..."
					 sleep 2
					 echo 
					 
					 echo -e "Fazendo o backup do arquivo smb.conf, aguarde..."
					 #Fazendo o backup do arquivo de configuração smb.conf
					 mv -v /etc/samba/smb.conf /etc/samba/smb.conf.old >> $LOG
					 echo -e "Backup feito com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Promovendo o controlador de domínio do SAMBA 4, aguarde..."
					 #Iniciando o processo de promoção do servidor utilizando o comando samba-tool domain provision
					 samba-tool domain provision --realm=$REALM --domain=$DOMAIN --server-role=$ROLE --dns-backend=$DNS --adminpass=$PASSWORD \
					 --function-level=$LEVEL --site=$SITE --host-ip=$IP --option="interfaces = lo eth0" --option="bind interfaces only = yes" \
					 --option="allow dns updates = nonsecure and secure" --option="dns forwarder = $IP" \
					 --option="winbind use default domain = yes" --option="winbind enum users  = yes" --option="winbind enum groups = yes" \
					 --option="winbind refresh tickets = yes" --option="server signing = auto" --option="vfs objects = acl_xattr" \
					 --option="map acl inherit = yes" --option="store dos attributes = yes" --option="client use spnego = no" \
					 --option="use spnego = no" --option="client use spnego principal = no" --use-rfc2307 --use-xattrs=yes &>> $LOG
					 echo -e "Promoção do Servidor SAMBA 4 feita com Sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Verificando as informações do Controlador de Domínio do SAMBA 4, aguarde..."
					 echo
					 #Verificando informações do log domínio.
					 tail -n6 $LOG | head -n6
					 echo
					 echo -e "Servidor SAMBA 4 promovido como Controlador de Domínio com Sucesso!!!"
					 echo -e "Pressione <Enter> para continuar com o script"
					 read
					 sleep 2
					 clear
					 echo ============================================================ >> $LOG
					 
					 echo -e "Desabilitando a expiração da senha do usuário Administrator, aguarde..."
					 #Desativando a expiração da senha do usuário administrator
					 samba-tool user setexpiry administrator --noexpiry &>> $LOG
					 echo -e "Senha desabilitada com sucesso!!!, pressione <Enter> continuando com o script"
					 read
					 sleep 2
					 clear
					 
					 echo -e "Editando o arquivo NAMED.CONF, aguarde..."
					 echo -e "Acrescentar a linha: include "/var/lib/samba/private/named.conf" no final arquivo named.conf"
					 echo -e "Pressione <Enter> para editar o arquivo"
					 echo 
					 read
					 
					 echo -e "Fazendo o backup do arquivo named.conf, aguarde..."
					 #Fazendo o backup do arquivo de configuração named.conf
					 mv -v /etc/bind/named.conf /etc/bind/named.conf.old >> $LOG
					 echo -e "Backup feito com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Atualizando o arquivo named.conf, aguarde..."
					 #Copiando o arquivo de configuração named.conf
					 cp -v conf/named.conf /etc/bind/named.conf >> $LOG
					 echo -e "Arquivo atualizado com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 #Editando o arquivo de configuração named.conf
					 vim /etc/bind/named.conf +19
					 
					 echo -e "NAMED.CONF atualizado com sucesso!!!, pressione <Enter> para continuando com o script"
					 read
					 sleep 2
					 clear
					 echo ============================================================ >> $LOG

					 echo -e "Editando o arquivo NAMED.CONF.OPTIONS"
					 echo -e "Acrescentar as linhas depois de: listen-on-v6 { any; };"
					 echo -e "Pressione <Enter> para editar o arquivo"
					 echo 
					 read
					 
					 echo -e "Criando o diretório e arquivo de Estatísticas do Bind9 RNDC, aguarde..."
					 #Criando o diretório para estatísticas do bind9 rndc
					 mkdir -v /var/log/named >> $LOG
					 #Criando o arquivo de estatísticas do bind9 rndc
					 touch /var/log/named/named.stats
					 #Alterando as permissões do diretório e do arquivo
					 chown -Rv bind.bind /var/log/named/ >> $LOG
					 echo -e "Diretório e arquivo criado com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Fazendo o backup do arquivo named.conf.options, aguarde..."
					 #Fazendo o backup do arquivo de configuração named.conf.options
					 mv -v /etc/bind/named.conf.options /etc/bind/named.conf.options.old >> $LOG
					 echo -e "Backup feito com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Atualizando o arquivo named.conf.options, aguarde..."
					 #Copiando o arquivo de configuração named.conf.options
					 cp -v conf/named.conf.options /etc/bind/named.conf.options >> $LOG
					 echo -e "Arquivo atualizado com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 #Editando o arquivo de configuração named.conf.options
					 vim /etc/bind/named.conf.options
					 
					 echo -e "NAMED.CONF.OPTIONS atualizado com sucesso!!!, pressione <Enter> para continuando com o script"
					 read
					 sleep 2
					 clear
					 echo ============================================================ >> $LOG

					 echo -e "Editando o arquivo NAMED.CONF.LOCAL"
					 echo -e "Após a modificação do arquivo named.conf.local será gerado a chave novamente"
					 echo -e "Será criado um Link Simbolico para o serviço do DHCP utilizar a mesma chave"
					 echo -e "Acrescentar a linha: include "/etc/bind/rndc.key" no final do arquivo"
					 echo -e "Pressione <Enter> para editar o arquivo"
					 echo
					 read
					 
					 echo -e "Fazendo o backup do arquivo named.conf.local, aguarde..."
					 #Fazendo o backup do arquivo de configuração named.conf.local
					 mv -v /etc/bind/named.conf.local /etc/bind/named.conf.local.old >> $LOG
					 echo -e "Backup feito com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Atualizando o arquivo named.conf.local, aguarde..."
					 #Copiando o arquivo de confguração named.conf.local
					 cp -v conf/named.conf.local /etc/bind/named.conf.local >> $LOG
					 echo -e "Arquivo atualizado com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 #Editando o arquivo de configuração named.conf.local
					 vim /etc/bind/named.conf.local +15
					 echo
					 
					 #Recurso de geração de chaves do Bind, recurso desativado em: 19/07/2016, falha na geração
					 #rndc-confgen -a
					 
					 echo -e "Alterando as permissões do arquivo rndc.key, aguarde..."
					 #Alterando as permissões de dono e grupo do arquivo de chaves rndc.key
					 chown -v root:bind /etc/bind/rndc.key >> $LOG
					 #Alterando as permissões de acesso do arquivo de chaves rndc.key
					 chmod -v 640 /etc/bind/rndc.key >> $LOG
					 #Criando um Link Simbólico do arquivo de chaves rndc.key
					 ln -v /etc/bind/rndc.key /etc/dhcp/ddns-keys/rndc.key >> $LOG
					 #Alterando as permissões de dono e grupo do link simbólico do arquivo de chaves rndc.key
					 chown -v root:bind /etc/dhcp/ddns-keys/rndc.key >> $LOG
					 echo -e "Permissões alteradas com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "NAMED.CONF.LOCAL atualizado com sucesso!!!, pressione <Enter> para continuando com o script"
					 read
					 sleep 2
					 clear
					 echo ============================================================ >> $LOG

					 echo -e "Editando o arquivo APPARMOR.D para permitir o BIND9 acessar os arquivos do SAMBA4"
					 echo -e "Acrescentar as informações no final do arquivo"
					 echo -e "Pressione <Enter> para editar o arquivo"
					 echo 
					 read
					 
					 echo -e "Fazendo o backup do arquivo usr.sbin.named, aguarde..."
					 #Fazendo o backup do arquivo de configuração usr.sbin.named
					 mv -v /etc/apparmor.d/local/usr.sbin.named /etc/apparmor.d/local/usr.sbin.named.old >> $LOG
					 echo -e "Backup feito com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Atualizando o arquivo usr.sbin.named, aguarde..."
					 #Copiando o arquivo de configuração do usr.sbi.named
					 cp -v conf/usr.sbin.named /etc/apparmor.d/local/usr.sbin.named >> $LOG
					 echo -e "Arquivo atualizado com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 #Editando o arquivo de configuração usr.sbi.named
					 vim /etc/apparmor.d/local/usr.sbin.named
					 echo
					 
					 echo -e "Alterando as permissões do arquivo dns.keytab e named.conf, aguarde..."
					 #Alterando as permissões de dono e grupo do arquivo de chaves dns.keytab
					 chown -v bind:bind /var/lib/samba/private/dns.keytab >> $LOG
					 #Alterando as permissões de acesso ao grupo no arquivo de chves dns.keytab
					 chmod -v g+r /var/lib/samba/private/dns.keytab >> $LOG
					 #Alterando o dono grupo do arquivo de configuração do SAMBA4 named.conf
					 chown -v bind:bind /var/lib/samba/private/named.conf >> $LOG
					 echo -e "Permissões alteradas com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "APPARMOR.D atualizado com sucesso!!!, pressione <Enter> para continuando com o script"
					 read
					 sleep 2
					 clear
					 echo ============================================================ >> $LOG

					 echo -e "Editando o arquivo SYSCTL.CONF para o servidor: `hostname`"
					 echo -e "Acrescentar no final do arquivo a linha: fs.file-max=500000"
					 echo -e "Aumentar o número de arquivos que o BIND9 vai gerenciar"
					 echo -e "Pressione <Enter> para editar o arquivo"
					 read
					 
					 echo -e "Fazendo o backup do arquivo sysctl.conf, aguarde..."
					 #Fazendo o backup do arquivo de configuração sysctl.conf
					 mv -v /etc/sysctl.conf /etc/sysctl.conf.old >> $LOG
					 echo -e "Backup feito com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Atualizando o arquivo sysctl.conf, aguarde..."
					 #Copiando o arquivo de configuração sysctl.conf
					 cp -v conf/sysctl.conf /etc/sysctl.conf >> $LOG
					 echo -e "Arquivo atualizado com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 #Editando o arquivo de configuração sysctl.conf
					 vim /etc/sysctl.conf +73
					 
					 echo -e "SYSCTL.CONF atualizado com sucesso!!!, pressione <Enter> para continuando com o script!!!!"
					 read
					 sleep 2
					 clear
					 echo ============================================================ >> $LOG


					 echo -e "Editando o arquivo LIMITS.CONF para o servidor: `hostname`"
					 echo -e "Acrescentar no final do arquivo a linha: *		-	nofile		400000"
					 echo -e "Pressione <Enter> para editar o arquivo"
					 read
					 
					 echo -e "Fazendo o backup do arquivo limits.conf, aguarde..."
					 #Fazendo o backup do arquivo de configuração limits.conf
					 mv -v /etc/security/limits.conf /etc/security/limits.conf.old >> $LOG
					 echo -e "Backup feito com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 echo -e "Atualizando o arquivo limits.conf, aguarde..."
					 #Copiando o arquivo de configuração limits.conf
					 cp -v conf/limits.conf /etc/security/limits.conf >>$LOG
					 echo -e "Arquivo atualizado com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 #Editando o arquivo de configuração limits.conf
					 vim /etc/security/limits.conf +70

					 echo -e "LIMITS.CONF atualizado com sucesso!!!, pressione <Enter> para continuando com o script!!!!"
					 read
					 sleep 2
					 clear
					 echo ============================================================ >> $LOG

					 echo -e "Editando o arquivo NAMED.CONF do SAMBA 4"
					 echo -e "Verificar as linhas referente a versão do BIND instalado"
					 echo
					 echo -e "Versão do BIND instalado: `named -v`"
					 echo
					 echo -e "Pressione <Enter> para editar o arquivo"
					 echo 
					 read
					 
					 echo -e "Fazendo o backup do arquivo named.conf, aguarde..."
					 cp -v /var/lib/samba/private/named.conf /var/lib/samba/private/named.conf.old &>> $LOG
					 echo -e "Backup feito com sucesso!!!, continuando o script..."
					 sleep 2
					 echo
					 
					 #Editando o arquivo de configurando do SAMBA 4 named.conf
					 vim /var/lib/samba/private/named.conf +20

					 echo -e "NAMED.CONF do SAMBA4 atualizado com sucesso!!!, pressione <Enter> para continuando com o script"
					 read
					 sleep 2
					 clear
					 
					 echo -e "Fim do Script-06.sh em: `date`" >> $LOG
					 echo ============================================================ >> $LOG

					 echo
					 echo -e "Configuração do SAMBA 4 como Controlador de Domínio feito com Sucesso!!!!!"
					 echo
					 # Script para calcular o tempo gasto para a execução do script-06.sh
						DATAFINAL=`date +%s`
						SOMA=`expr $DATAFINAL - $DATAINICIAL`
						RESULTADO=`expr 10800 + $SOMA`
						TEMPO=`date -d @$RESULTADO +%H:%M:%S`
					 echo -e "Tempo gasto para execução do script-06.sh: $TEMPO"
					 echo -e "Pressione <Enter> para concluir o processo e reiniciar o servidor."
					 sleep 3
					 read
					 reboot
					 else
						 echo -e "Versão do Kernel: $KERNEL não homologada para esse script, versão: >= 4.4 "
						 echo -e "Pressione <Enter> para finalizar o script"
					 read
			fi
		else
			 echo -e "Distribuição GNU/Linux: `lsb_release -is` não homologada para esse script, versão: $UBUNTU"
			 echo -e "Pressione <Enter> para finalizar o script"
			read
		fi
else
	 echo -e "Usuário não é ROOT, execute o comando com a opção: sudo -i <Enter> depois digite a senha do usuário `whoami`"
	 echo -e "Pressione <Enter> para finalizar o script"
	read
fi
